plugins/outputs/postgresql/postgresql.go

```diff
@@ -0,0 +1,424 @@
+  ## e.g. PGPASSWORD, PGHOST, PGUSER, PGDATABASE 
+  ## all supported vars here: https://www.postgresql.org/docs/current/libpq-envars.html
+  ##
+  ## Non-standard parameters:
+  ##   pool_max_conns (default: 1) - Maximum size of connection pool for parallel (per-batch per-table) inserts.
+  ##   pool_min_conns (default: 0) - Minimum size of connection pool.
+  ##   pool_max_conn_lifetime (default: 0s) - Maximum age of a connection before closing.
+  ##   pool_max_conn_idle_time (default: 0s) - Maximum idle time of a connection before closing.
+  ##   pool_health_check_period (default: 0s) - Duration between health checks on idle connections.
+  ##
+  ## Without the dbname parameter, the driver will default to a database
+  ## with the same name as the user. This dbname is just for instantiating a
+  ## connection with the server and doesn't restrict the databases we are trying
+  ## to grab metrics for.
+  ##
+  #connection = "host=localhost user=postgres sslmode=verify-full"
+
+  ## Postgres schema to use.
+  schema = "public"
+
+  ## Store tags as foreign keys in the metrics table. Default is false.
+  tags_as_foreign_keys = false
+
+  ## Suffix to append to table name (measurement name) for the foreign tag table.
+  tag_table_suffix = "_tag"
+
+  ## Deny inserting metrics if the foreign tag can't be inserted.
+  foreign_tag_constraint = false
+
+  ## Store all tags as a JSONB object in a single 'tags' column.
+  tags_as_jsonb = false
+
+  ## Store all fields as a JSONB object in a single 'fields' column.
+  fields_as_jsonb = false
+
+  ## Templated statements to execute when creating a new table.
+  create_templates = [
+    '''CREATE TABLE {{.table}} ({{.columns}})''',
+  ]
+
+  ## Templated statements to execute when adding columns to a table.
+  ## Set to an empty list to disable. Points containing tags for which there is no column will be skipped. Points
+  ## containing fields for which there is no column will have the field omitted.
+  add_column_templates = [
+    '''ALTER TABLE {{.table}} ADD COLUMN IF NOT EXISTS {{.columns|join ", ADD COLUMN IF NOT EXISTS "}}''',
+  ]
+
+  ## Templated statements to execute when creating a new tag table.
+  tag_table_create_templates = [
+    '''CREATE TABLE {{.table}} ({{.columns}}, PRIMARY KEY (tag_id))''',
+  ]
+
+  ## Templated statements to execute when adding columns to a tag table.
+  ## Set to an empty list to disable. Points containing tags for which there is no column will be skipped.
+  tag_table_add_column_templates = [
+    '''ALTER TABLE {{.table}} ADD COLUMN IF NOT EXISTS {{.columns|join ", ADD COLUMN IF NOT EXISTS "}}''',
+  ]
+
+  ## When using pool_max_conns>1, an a temporary error occurs, the query is retried with an incremental backoff. This
+  ## controls the maximum backoff duration.
+  retry_max_backoff = "15s"
+
+  ## Enable & set the log level for the Postgres driver.
+  # log_level = "info" # trace, debug, info, warn, error, none
```

Comment:
Not sure we need this. How about setting it to `warn` for normal operation and to `debug` if telegraf is in debugging mode? I'm not sure if we already have this information about telegraf yet, but if not we probably should add it.

Reply:
This is for the pgx. Setting it adds overhead, which is why it's not set by default. However if the user is trying to define a template that doesn't work, being able to see the SQL that's getting executed (and whatever else the pgx driver is doing) can be useful.

Reply:
That's how I understood it. My question is if we can work with two modes, e.g. "production" and "debug" and couple this to the telegraf debug flag? So in "normal"/"production" operation you might want to see all errors or warnings for the pgx driver, and in "debug" you want to find out what is going on e.g. to see what SQL is executed.

Reply:
I would prefer to keep multiple log levels available. Some people might want to run with warnings enabled, and some might prefer errors only. Similar between `trace` and `debug`.

However I don't have any issue if we want to make it so that it defaults to `error` if in normal mode, and default to `debug` if in debug mode. Is there a way to discover which mode telegraf is running in? `Init()` doesn't receive any parameters, so I don't see how.